service/networking.py

```diff
@@ -1,8 +1,303 @@
-from atmosphere.settings import ATMOSPHERE_PRIVATE_KEYFILE
+"""
+Atmosphere network
+
+This file will help a driver infer the strategy to use when creating or destroying a "project network".
+"""
+import random
+
+from neutronclient.common.exceptions import NeutronClientException, NotFound
+
+from atmosphere import settings
+from threepio import logger
 
 
 def _generate_ssh_kwargs(timeout=120):
     kwargs = {}
-    kwargs.update({'ssh_key': ATMOSPHERE_PRIVATE_KEYFILE})
+    kwargs.update({'ssh_key': settings.ATMOSPHERE_PRIVATE_KEYFILE})
     kwargs.update({'timeout': timeout})
     return kwargs
+
+def _get_unique_id(userid):
+    if 'iplantauth.authBackends.LDAPLoginBackend' in settings.AUTHENTICATION_BACKENDS:
+        from iplantauth.protocol.ldap import _get_uid_number
+        return _get_uid_number(userid)
+    else:
+        return _get_random_uid(userid)
+
+def _get_random_uid(userid):
+    """
+    Given a string (Username) return a value < MAX_SUBNET
+    """
+    MAX_SUBNET = 4064
+    return int(random.uniform(1, MAX_SUBNET))
+
+
+def get_topology_cls(topology_name):
+    if topology_name == ExternalNetwork.name:
+        return ExternalNetwork
+    elif topology_name == ExternalRouter.name:
+        return ExternalRouter
+    else:
+        raise Exception("Unknown topology name %s" % topology_name)
+
+
+def get_ranges(uid_number, inc=0):
+    """
+    Return two block ranges to be used to create subnets for
+    Atmosphere users.
+
+    NOTE: If you change MAX_SUBNET then you should likely change
+    the related math.
+    """
+    MAX_SUBNET = 4064  # Note 16 * 256
+    n = uid_number % MAX_SUBNET
+
+    #16-31
+    block1 = (n + inc) % 16 + 16
+
+    #1-254
+    block2 = ((n + inc) / 16) % 254 + 1
```

Comment:
This whole thing needs re-written. its a port from https://github.com/iPlantCollaborativeOpenSource/rtwo/blob/0.3/rtwo/drivers/openstack_network.py#L456-L497